tests/e2e/tests/multicluster/split_horizon_test.go

```diff
@@ -225,10 +227,50 @@ type deployableConfig struct {
 	// (e.g the default global authentication policy) and need to be modified for tests.
 	Removes    []resource
 	applied    []string
+	appliedOut []string
 	removed    []string
 	kubeconfig string
 }
 
+func (c *deployableConfig) Wait(istioctls []*framework.Istioctl) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	g, _ := errgroup.WithContext(ctx)
+	// Wait until the deployments are ready
+
+	g.Go(func() error {return util.CheckDeployments(c.Namespace, maxDeploymentTimeout, c.kubeconfig)})
+
+	for _, singleOut := range c.appliedOut {
+		for _, line := range strings.Split(singleOut, "\n") {
+			// output line is in format:
+			// FullyQualifiedType/name [updated|create|unchanged]
```

Comment:
when we make this generic, make it so it's applyAndWait then we don't need to parse the apply output and can instead read the input?

Reply:
By 'read the input', do you mean that we should parse the incoming yaml files for object types and names?  That seems more complicated than parsing the apply command's output... is there another advantage I'm not seeing?

Reply:
most of my comments here are in the context of the newer test framework that is used by tests under test/integration. These support local and kubernetes testing, so they aren't doing things like shelling out to kubectl. They already do the parsing, so we can probably just hook into that logic

Reply:
interesting.  I'll have a look.  How do those tests handle calling istioctl, or accessing the pilot *z endpoints?